openid-4-verifiable-presentations-1_0.md

```diff
@@ -559,11 +565,276 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+Credential. 
+
+It contains the following properties:
+
+`id`:
+: REQUIRED. A string identifying the Credential in the response and, if provided,
+the constraints in `valid_credential_sets`. The value MUST be a non-empty string
+consisting of alphanumeric, underscore (`_`) or hyphen (`-`) characters.
+Within the Authorization Request, the same `id` MUST NOT
+be present more than once.
+
+`format`:
+: REQUIRED. A string that specifies the format of the requested
+Verifiable Credential. Valid Credential Format Identifier values are defined in
+Appendix A of [@!OpenID.VCI]. The value of this property MUST be one of the
+supported Credential Formats as defined in the Wallet's metadata.
+
+`expect_meta`: 
+: OPTIONAL. An object defining additional properties requested by the Verifier that
+apply to the metadata and validity data of the Credential. The properties of
+this object are defined per Credential Format in (#format_specific_properties). If omitted,
+no specific constraints are placed on the metadata or validity of the requested
+Credential.
+
+`expect_claims`:
+: OPTIONAL. A non-empty array of objects as defined in (#claims_query) that specifies
+claims in the requested Credentials.
+
+`valid_claim_sets`:
+: OPTIONAL. A non-empty array containing arrays of identifiers for elements in
+`expect_claims`. The identifier MAY be postfixed by `?`, indicating that
+delivery of the the respective claim is OPTIONAL.  
+
+## Claims Query {#claims_query}
+
+Each entry in `expect_claims` MUST be an object with the following properties:
+
+`id`:
+: REQUIRED if `valid_claim_sets` is present; OPTIONAL otherwise. A string
+identifying the particular claim. The value MUST be a non-empty string
+consisting of alphanumeric, underscore (`_`) or hyphen (`-`) characters.
+Within the particular `expected_claims` array, the same `id` MUST NOT
+be present more than once.
+
+`path`:
+: REQUIRED if the Credential Format uses a JSON-based claims structure; MUST NOT
+be present otherwise. The value MUST be a claims path pointer that specifies the path to the claim
+within the Verifiable Credential, as defined in (#claims_path_pointer).
+
+`namespace`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the namespace of the claim
+within the Verifiable Credential, e.g., `org.iso.18013.5.1`.
+
+`claim_name`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the name of the claim within the provided namespace
+in the Verifiable Credential, e.g., `first_name`.
+
+`value`:
+: OPTIONAL. A string, integer or boolean value that specifies the expected value of the claim. If the
+`value` property is present, the Wallet MUST return the claim only if the type and value
+of the claim match the type and value specified in the query.
+
+`values`:
```

Comment:
Would suggest only having this option if we keep the match by value feature as you can accomplish the functionality provided by the `value` parameter with an array of 1 using the `values` parameter.

Reply:
Also at the very least we would need to ensure value and values can both be present in a request.

Reply:
`value` and `values` are both defined in OIDC, some people might expect both to exist.

Reply:
I'll add clauses on mutual exclusivity of `value` and `values`

Reply:
I dont think we should prioritise consistency with OIDC here, one parameter for this functionality would lead to simpler and more robust implementations, requiring mutual exclusivity in maps/objects can be a pain and liable to be a check certain implementations could miss. For example what should an implementation do if it encounters both, should it error?

Reply:
I'd like to hear what @selfissued thinks on this one.

Reply:
+1 to only keeping `values`.